lib.rs

```diff
@@ -1037,21 +1218,25 @@ impl<A, B> ArcUnion<A, B> {
     }
 
     /// Creates an `ArcUnion` from an instance of the first type.
+    #[inline]
     pub fn from_first(other: Arc<A>) -> Self {
         unsafe { Self::new(Arc::into_raw(other) as *mut _) }
     }
 
     /// Creates an `ArcUnion` from an instance of the second type.
+    #[inline]
     pub fn from_second(other: Arc<B>) -> Self {
         unsafe { Self::new(((Arc::into_raw(other) as usize) | 0x1) as *mut _) }
     }
 
     /// Returns true if this `ArcUnion` contains the first type.
+    #[inline]
     pub fn is_first(&self) -> bool {
         self.p.as_ptr() as usize & 0x1 == 0
```

Comment:
I guess this works because no matter what `A` and `B` are, ArcInner<A>/ArcInner<B> always is aligned to at least `align_of::<usize>()`. Neat.